datafusion/common/src/scalar/mod.rs

```diff
@@ -1773,6 +1773,7 @@ impl ScalarValue {
             }
             DataType::List(_)
             | DataType::LargeList(_)
+            | DataType::Map(_, _)
```

Comment:
this works at least for the test case, given it re-uses arrow::compute::concat I'd hope it does the right thing overall

Reply:
this is fine I think because `ScalarValue::Map` is implemented as a 1 row array